packages/domain/source/shared/todos/rules/unique-todo.js

```diff
@@ -0,0 +1,29 @@
+Space.domain.Rule.extend('Todos.UniqueTodoRule', {
+
+  mixin: [
+    Space.messaging.CommandSending
+  ],
+
+  enforceMap() {
+    return [{
+      'Todos.CreateTodo': this._onCreateTodo
+    }]
+  },
+
+  _onCreateTodo(command) {
+    console.log(
+      ['rule', 'before'], 'Todos.UniqueTodoRule', command.toString(),
+      ['S', 'C']
+    )
+
+    this.send(
+      new Todos.ValidateTodo({title: command.title}), function(err, result) {
+        if (result === false) {
```

Comment:
Yes I'm calling them queries in my code. I think it would be nice to have syntax sugar like `Space.messaging.Query` that is defined as Commands - with fields for validation.

In this scenario sure, however for most - we can't expose `repository` with `find` methods to client. Like imagine giving access to whole collection of email addresses just to search on client if current - is not yet used for registration.

Thats why there is trip to server without exposing directly persistence. And the goal behind this pattern was to have a very easy to set unique validation that can be shared on client and server.

In this case its not necessary - this could end up on client just like question to `Todos.TodoLists` if its taken, without trip to server. I'm using this as simple playground and concept showoff.